### ISSUE TITLE
Tank dry mass not properly calculated

### ISSUE THREAD

**Issue by RobertoBiundo** [KSP-RO/RealFuels#343]
I have tanken the liberty to check the dry mass calculation of tanks created with procedural tanks and have found that they are not properly calculated. As an example ill start with a WAC corporal "alike" tank (cilindrical)

The height of my tank is 4.8m and width 30.5cm (diameter).
This gives in the game a volume of 263L and a dry mass of 102Kg. Ok, fair enough, we will use this as a point of reference.

Lets start checking this. The volume of a cilinder is:
`V = pi * r^2 * h`
And the area is
`A = ( 2 * pi * r^2 ) + ( 2 * pi * r * h )`
For the provided parameters above this gives us 350.7L -> With a 75% ( V * 0.75 ) tank utilization this is indeed 263L
And the area would be 47454 cm^2 -> 4.74 m^2. (This is the amount of material needed. 4.74m^2 of metal)

So far so good. Now lets create an optimized tank that hauls the same volume but with the minimal possible Area (thus reducing the dry mass both materials being equal).

This means we need to optimize for V and A
We now
`V = pi * r^2 * h`
and
`A = ( 2 * pi * r^2 ) + ( 2 * pi * r * h )`

form the first formula we get that 
`h = V / ( pi * r^2 )`

using this in the formula for area we get
`A = ( 2 * pi * r^2 ) + ( 2 * V / r )`

So far so good. We want now to minimize this formula so we get the derivate of dA/dr
`dA/dr = ( 4 * pi * r^2 ) - ( 2 * V / r^2 )`

Calculating where the derivate is 0 to get the min point
`( 4 * pi * r^2 ) - ( 2 * V / r^2 ) = 0`
we get
`r = cubic_sqrt( ( 2 * V ) / ( 4 * pi )`

For the value of 350L ( to account for the 75% utilization ) we get that r should be 38.19cm ( diameter of 76.38cm ) and the height should be ( 76.38 cm ) <= AS EXPECTED

Now this also yeild an Area of 27491 cm^2 -> 2.74 m^2  (This is the amount of material needed. 2.74m^2 of metal)

**TL;DR**

NOW. We have crated a new tank that holds the same but with a lot less material. Granted, the aerodynamics are horrible, but it is lighter. Is it?........well....according to the game, no. both tanks have a dry mas of 102Kg. Actually any tank, on any configuration that you create that hauls 263L will have a dry mass of 102Kg no matter what. Quite unrealistic in my optinion. It would be amazing to not only minimize the formulas above but be able to include aerodynamics in the mix as well. This would allow you to get the best aerodynamic possible for the least weight possible.

The issue seems to plague all other tank types as well (i,e. cone type tanks suffer from the same). And this is a shame.
Now, I know i'm being annoying as hell and that there are more pressing bugs. But in a game mode that has principia calculating the impact of all gravity forces on a ship is a shame that the ship itself has a weight that is wrongly calculated. 

Would someone please fix this?, or point me in the right direction on the code so that I can fix it. I know nothing about the codebase and don't want to spend the rest of my month going over the whole thing. But I am a decent C# developer and if you point me to the correct part of the src ill pass on a PR your way.

**Comment by Capkirk123** [comment]
Transferred to the correct repo. I believe the code in question is [here](https://github.com/KSP-RO/RealFuels/blob/master/Source/Tanks/FuelTank.cs)

**Comment by RobertoBiundo** [comment]
Great. That was quick for not knowing the source code. I can work on this o
and propose an improvement.
Whish me luck. Time to dig into a physics rabbit hole

**Comment by siimav** [comment]
Haven't heard anything in a while. Can reopen the issue if there's anything further to discuss.
